src/ResourceManager/Common/Commands.ResourceManager.Common/RMProfileClient.cs

```diff
@@ -297,14 +308,21 @@ private AzureEnvironment MergeEnvironmentProperties(AzureEnvironment environment
                         var subscriptions = subscriptionClient.Subscriptions.List().Subscriptions;
                         if (subscriptions != null && subscriptions.Any())
                         {
-                            if (subscriptions.Count > 1)
+                            if (subscriptionName != null)
+                            {
+                                subscriptionFromServer = subscriptions.First(s => s.DisplayName.Equals(subscriptionName, StringComparison.OrdinalIgnoreCase));
```

Comment:
This will produce a bad exception when there are no matches. Do FirstOrDefault instead. If `subscriptionFromServer` is null it will be cought on line 334